### ISSUE TITLE
bug: Glitchy and inconsistent headerbars in some GTK 3 apps

### ISSUE THREAD

**Issue by dharmik2319** [GradienceTeam/Gradience#779]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

When I apply a theme generated by the monet engine, the libadwaita and GTK 4 apps look and behave flawlessly. But, some GTK 3 apps (with adw-gtk3) have glitchy headerbars due to transparency. In fact, all of them show inconsistent behavior. gnome-terminal has a completely transparent headerbar when in focus, otherwise it becomes opaque. VLC always has a transparent headerbar regardless of focus. gnome-control-center or the Settings app and gnome-tweaks have the glitchy headerbars with severe flickering. nautilus looks and behaves like a GTK 4 app even though it is GTK 3. GTK 4 and libadwaita apps dont have transparency in their headerbars at all, which is expected.

### To Reproduce

1. Apply [this theme] (https://github.com/GradienceTeam/Gradience/files/11645517/wal.txt) (renamed to txt to support github upload, you will need to rename to json).
2. Make sure to apply to GTK 3 apps as well
3. check the apps mentioned above

### Expected behavior

All GTK 3 apps have consistent behaviour, among themselves and with GTK 4 and libadwaita as well.

### Screenshots

![Screenshot from 2023-06-04 17-36-02](https://github.com/GradienceTeam/Gradience/assets/126943942/510a7c7b-1ac0-47f6-8235-3cca6c798bac)
Nautilus looking normal even though it is GTK 3
![Screenshot from 2023-06-04 17-36-11](https://github.com/GradienceTeam/Gradience/assets/126943942/899458da-3d99-4835-b426-a7c6171a6834)
gnome-terminal in focus having transparent headerbar
![Screenshot from 2023-06-04 17-36-22](https://github.com/GradienceTeam/Gradience/assets/126943942/2d9d053a-10b9-4e10-9c08-c5f5c26daeb9)
gnome-terminal headerbar losing transparency upon losing focus
![Screenshot from 2023-06-04 17-36-46](https://github.com/GradienceTeam/Gradience/assets/126943942/5cad6063-7c9a-4bc8-a57a-eb4ce7901deb)
![Screenshot from 2023-06-04 17-36-54](https://github.com/GradienceTeam/Gradience/assets/126943942/7f7a6662-a107-4f1d-8353-459b133a6a4c)
![Screenshot from 2023-06-04 17-37-09](https://github.com/GradienceTeam/Gradience/assets/126943942/49315e7d-c4ac-4165-9c33-75e96c4822cb)
![Screenshot from 2023-06-04 17-37-29](https://github.com/GradienceTeam/Gradience/assets/126943942/5408b24b-461f-4962-9fc8-489e79eebf16)
Various GTK 3 apps showing flickering and glitchiness in their transparency, which they lose upon losing focus
![image](https://github.com/GradienceTeam/Gradience/assets/126943942/79b250b3-cccd-4fbb-b720-ceba06f812eb)
Actual libadwaita/GTK 4 apps, without transparency

### OS

`cat /etc/os-release`: NAME="Pop!_OS" VERSION="22.04 LTS" ID=pop ID_LIKE="ubuntu debian" PRETTY_NAME="Pop!_OS 22.04 LTS" VERSION_ID="22.04" HOME_URL="https://pop.system76.com" SUPPORT_URL="https://support.system76.com" BUG_REPORT_URL="https://github.com/pop-os/pop/issues" PRIVACY_POLICY_URL="https://system76.com/privacy" VERSION_CODENAME=jammy UBUNTU_CODENAME=jammy LOGO=distributor-logo-pop-os

### DE/WM version

GNOME Shell 42.5

### Version

0.4.1

### Installation method

Flatpak from Flathub

### Enabled system extensions (GNOME specific)

clipboard-indicator@tudmotu.com
dash-to-dock-cosmic-@halfmexicanhalfamazing@gmail.com
emoji-selector@maestroschan.fr
compiz-windows-effect@hermes83.github.com
cosmic-workspaces@system76.com
pop-cosmic@system76.com
pop-shell@system76.com
system76-power@system76.com
screenshot-window-sizer@gnome-shell-extensions.gcampax.github.com
user-theme@gnome-shell-extensions.gcampax.github.com

### Additional context

N/A

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by daudix-UFO** [comment]
This is known issue, it happens because of alpha value in headerbar color, you can remove last digits from it to make it opaque

The fix will come in next release

**Comment by dharmik2319** [comment]
Thank you very much for the quick reply. I will try doing that and close afterwards.

**Comment by daudix-UFO** [comment]
No need in closing, we can close it when the fix will land

**Comment by dharmik2319** [comment]
>This is known issue, it happens because of alpha value in headerbar color, you can remove last digits from it to make it opaque

I did as you mentioned (see [here](https://github.com/GradienceTeam/Gradience/files/11651500/gtk3.txt)), but now, all the windows have headerbars with extremely light colors, which messes up the contrast.

![Screenshot from 2023-06-05 15-57-02](https://github.com/GradienceTeam/Gradience/assets/126943942/8097b37f-d904-4955-909b-51c174155cde)

![Screenshot from 2023-06-05 15-57-11](https://github.com/GradienceTeam/Gradience/assets/126943942/ecbd9df7-278b-45ac-a8e0-c21dfa85e088)
![Screenshot from 2023-06-05 15-57-28](https://github.com/GradienceTeam/Gradience/assets/126943942/48039ef5-4239-40f6-bcba-bfbfc3dc00c9)
![Screenshot from 2023-06-05 15-57-32](https://github.com/GradienceTeam/Gradience/assets/126943942/bf706664-f0e5-4596-a1e6-986d851e89f3)
All of this, again happens only when the window is in focus, when not in focus it behaves fine.

**Comment by daudix-UFO** [comment]
Yeah, we used transparency to make color less bright, but as you see transparency breaks gtk3

**Comment by dharmik2319** [comment]
After some fiddling with color pickers, looks and behaves flawlessly!

Thank you VERY much for this app.

**Comment by dharmik2319** [comment]
If anyone is wondering what i did to fix it:

1. First of all, take a screenshot of the gradience window.
2. Open a color picker and copy the hex code for the header bar color, from the screenshot, if you try otherwise, it gains focus and things change
3. replace headerbar_bg_color to the copied hex code
4. for headerbar_border_color just change the rgba to rgb and delete the last number
5. headerbar_shade_color should be the same as window_bg_color
6. You're done.

**Comment by daudix-UFO** [comment]
Temporary fix landed in https://github.com/GradienceTeam/Gradience/commit/b55eecbdb4870e6636642037fc6b6233dfd708c2

**Comment by LeSeskyCar** [comment]
Even i have this issue

**Comment by daudix-UFO** [comment]
> Even i have this issue

We know dude, everyone does :)

**Comment by dharmik2319** [comment]
> If anyone is wondering what i did to fix it:
> 
> 1. First of all, take a screenshot of the gradience window in focus.
> 2. Open a color picker and copy the hex code for the header bar color, from the screenshot, if you try otherwise, it gains focus and things change
> 3. replace headerbar_bg_color to the copied hex code
> 4. for headerbar_border_color just change the rgba to rgb and delete the last number
> 5. headerbar_shade_color should be the same as window_bg_color
> 6. You're done.



> Even i have this issue

you can try this to fix it for now
